src/AppInstallerCLICore/ExecutionReporter.h

```diff
@@ -165,9 +168,20 @@ namespace AppInstaller::CLI::Execution
             m_progressSink = sink;
         }
 
+        void SetLevelEnabled(ReporterLevel reporterLevel, bool setEnabled = true)

```

Comment:
Passing `None` to this method has ambiguous meaning.  For instance, what do I want from `SetLevelEnabled(None, true)`?  Do I want no levels enabled?  And does `false` then mean I want no levels disabled?  If those are the cases, then the logic is wrong.  In fact, passing `None` will have no effect on the value of `m_enabledLevels` regardless of `setEnabled`.

I think the easy solution is a name change:
```suggestion
        void SetLevelMask(ReporterLevel reporterLevel, bool setEnabled = true)
```
To move the noun from "level" to "mask", which better indicates the bitrack and thus the meaning of what will happen when `None` is passed.  If you want to work "enabled" back in, that is probably fine too.

Reply:
In this case, I think the easiest solution is to remove `None` from the enum entirely, as it has no real purpose. It isn't necessary for any behaviors currently, and to turn off all reporting would be functionally the same as using `All` with `false`

Reply:
The only purpose I see for `None` is to initialize the field if we wanted to start with no levels enabled instead of all.  Removing it is a valid option, but I personally would lean towards keeping `None` and changing the name of the method.  Not a big sticking point though.